sklearn/preprocessing/label.py

```diff
@@ -889,16 +890,24 @@ def transform(self, y):
             A matrix such that `y_indicator[i, j] = 1` iff `classes_[j]` is in
             `y[i]`, and 0 otherwise.
         """
+        self._cached_dict = None
```

Comment:
Well, if you reset it here it will never be reused.
You should remove this line, and put it in `fit_transform`.

Reply:
ok got a bit confused with this caching :D